src/main/scala/cognite/spark/v1/FileContentRelation.scala

```diff
@@ -0,0 +1,141 @@
+import org.apache.spark.{Partition, TaskContext}
+import sttp.capabilities.WebSockets
+import sttp.capabilities.fs2.Fs2Streams
+import sttp.client3.asynchttpclient.SttpClientBackendFactory
+import sttp.client3.asynchttpclient.fs2.AsyncHttpClientFs2Backend
+import sttp.client3.{SttpBackend, UriContext, asStreamUnsafe, basicRequest}
+import sttp.model.Header
+
+trait WithSizeLimit {
+  val sizeLimit: Long
+}
+
+class FileContentRelation(config: RelationConfig, fileExternalId: String, inferSchema: Boolean)(
+    override val sqlContext: SQLContext)
+    extends BaseRelation
+    with TableScan
+    with PrunedFilteredScan
+    with Serializable
+    with WithSizeLimit {
+
+  override val sizeLimit: Long = 5 * FileUtils.ONE_GB
+
+  @transient lazy val client: GenericClient[IO] =
+    CdpConnector.clientFromConfig(config)
+
+  @transient private lazy val sttpFileContentStreamingBackendResource
+    : Resource[IO, SttpBackend[IO, Fs2Streams[IO] with WebSockets]] =
+    for {
+      dispatcher <- Dispatcher.parallel[IO]
+      backend <- Resource.make(
+        IO(
+          AsyncHttpClientFs2Backend
+            .usingClient[IO](SttpClientBackendFactory.create("file content download"), dispatcher))
+      )(backend => backend.close())
+    } yield backend
+
+  private lazy val dataFrame: DataFrame = createDataFrame(sparkSession = sqlContext.sparkSession)
+
+  override def schema: StructType =
+    dataFrame.schema
+
+  def createDataFrame(sparkSession: SparkSession): DataFrame = {
+    val rdd: RDD[String] = new RDD[String](sparkSession.sparkContext, Nil) with Serializable {
+
+      import cognite.spark.v1.CdpConnector.ioRuntime
+
+      val maxParallelism = 1
+      override def compute(split: Partition, context: TaskContext): Iterator[String] = {
+
+        val validUrl = for {
+          mimeType <- client.files.retrieveByExternalId(fileExternalId).map(_.mimeType)
+          downloadLink <- client.files
+            .downloadLink(FileDownloadExternalId(fileExternalId))
+            .map(_.downloadUrl)
+          _ <- IO.pure(
+            if (!downloadLink.startsWith("https")) {
+              throw new CdfSparkException("File storage is not using https protocol")
+            }
+          )
+          isFileWithinLimits <- isFileWithinLimits(downloadLink)
+        } yield {
+          if (mimeType.isDefined && !Seq(
+              "application/jsonlines",
+              "application/x-ndjson",
+              "application/jsonl").contains(mimeType.get))
+            throw new CdfSparkException("Wrong mimetype. Expects application/jsonlines")
+          if (!isFileWithinLimits)
+            throw new CdfSparkException(
+              f"File size above size limit, or file size header absent from head request")
+          downloadLink
+        }
+
+        StreamIterator(
+          Stream
+            .eval(validUrl)
+            .flatMap(readUrlContent),
+          maxParallelism,
```

Comment:
```suggestion
          maxParallelism * 2,
```
maybe 1->2, SdkV1Rdd does that

Reply:
Hmm, I'm going to read a bit more on what this does, I *assumed* it had to be 1 so we wouldn't do several requests (since only 1 executor can possibly hold the request at the same time?) but I am not very well informed here so I4ll come back with an answer

Reply:
it can be anything I think, just a buffer size, with some code comments that we're neve able to fill it up to big sizes (maybe because for other resources reads are slow compared to that stream adaptor processing)